From e3e34afa649123dd19da14f212e95e09532f2a0b Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Fri, 26 Jun 2026 22:35:15 +0800
Subject: [PATCH] AOSCOS: kcm: enable screen layout configurator
 unconditionally

The current UI disables layout settings when there is only one available
display, shifting all configurations to the top without appropriate top
margin, making the interface look broken at the first glance.

Make the KCM show layout settings unconditionally (when there is >= 1
display attached) to make it look normal again.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 kcm/ui/OutputPanel.qml | 6 +++---
 kcm/ui/main.qml        | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/kcm/ui/OutputPanel.qml b/kcm/ui/OutputPanel.qml
index e6dc8671..28fc924e 100644
--- a/kcm/ui/OutputPanel.qml
+++ b/kcm/ui/OutputPanel.qml
@@ -36,17 +36,17 @@ Kirigami.FormLayout {
     }
 
     RowLayout {
-        visible: kcm.primaryOutputSupported && root.enabledOutputs.count >= 2
+        visible: kcm.primaryOutputSupported
 
         QQC2.Button {
-            visible: root.enabledOutputs.count >= 3
+            visible: true
             text: i18n("Change Screen Priorities…")
             icon.name: "document-edit"
             onClicked: root.reorder();
         }
 
         QQC2.RadioButton {
-            visible: root.enabledOutputs.count === 2
+            visible: true
             text: i18n("Primary")
             checked: element.priority === 1
             onToggled: element.priority = 1
diff --git a/kcm/ui/main.qml b/kcm/ui/main.qml
index 518d1388..f9f83394 100644
--- a/kcm/ui/main.qml
+++ b/kcm/ui/main.qml
@@ -378,7 +378,7 @@ KCM.AbstractKCM {
                 }
             }
 
-            visible: kcm.multipleScreensAvailable
+            visible: true
 
             ScreenView {
                 id: screen
-- 
2.52.0

